docs/source/Getting-Started/Tutorials/Build-Enclave-App-Hello-World.rst

```diff
@@ -0,0 +1,257 @@
+
+Before jumping into the tutorial, please complete :doc:`Quick Start
+<../Running-Keystone-with-QEMU>`.
+
+
+Prerequisite
+------------------------------
+
+`musl-libc <https://www.musl-libc.org/>`_ is a lightweight standard library for replacing ``glibc``.
+The Eyrie runtime allows the enclave to be statically linked with musl, and supports a few standard
+functions such as ``printf``.
+
+Download source code in your work directory, and build musl with following commands.
+
+::
+
+	cd <your path>
+	git clone https://github.com/riscv/riscv-musl
+	cd riscv-musl
+	./configure --prefix=<install path (absolute path)> --disable-shared --enable-static
+	--host=riscv64-unknown-linux-gnu
+	make; make install
+
+``musl-gcc`` will be installed at ``<install path>``.
+
+Open ``sdk/examples/hello/eapp/Makefile`` and set ``CC`` to point ``musl-gcc``
+
+:: 
+
+	CC=<install path>/bin/musl-gcc
+
+
+Finally, set ``PATH`` to include RISC-V tools and ``KEYSTONE_SDK_DIR`` to point the
+absolute path to ``sdk`` directory.
+
+::
+
+	export PATH=$PATH:<path to RISC-V tools>
+	export KEYSTONE_SDK_DIR=<path to SDK>
+
+Let's take a look at the example provided in `Keystone SDK
+<https://github.com/keystone-enclave/keystone-sdk>`_.
+
+::
+
+	ls sdk/examples/hello
+
+You can find two directories and a script called ``vault.sh``
+
+vault.sh
+------------------------------
+
+``vault.sh`` is a sample script that builds the enclave.
+
+
+::
+
+	sdk/examples/hello/vault.sh
+
+The script will compile the eapp, the runtime, and the host.
+Take a look at the first few lines of the script.
+In most cases, changing the variables in the script will be sufficient to build any custom enclaves.
+
+.. code-block:: bash
+
+	################################################################
+	#                   Replace the variables                      #
+	################################################################
+	NAME=hello
+	VAULT_DIR=`dirname $0`
+	BUILD_COMMAND="make -C eapp && make -C host"
+	OUTPUT_DIR=$KEYSTONE_SDK_DIR/../buildroot_overlay/root/$NAME
+	EYRIE_DIR=$KEYSTONE_SDK_DIR/rts/eyrie
+	EYRIE_PLUGINS="freemem untrusted_io_syscall linux_syscall env_setup"
+	PACKAGE_FILES="eapp/hello \
+	               host/runner \
+	               $EYRIE_DIR/eyrie-rt"
+	PACKAGE_SCRIPT="./runner hello eyrie-rt"
+
+We will go through what each variable is for, and how the script builds the enclave.
+
+``NAME`` defines the name of the enclave. It will be used to generate the final enclave package.
+
+``VAULT_DIR`` is the path to the script. The script will change the directory to ``$VAULT_DIR``
+before running the build commands.
+
+``BUILD_COMMAND`` defines the build commands. This example has ``Makefile`` in each of the
+directory, so the script just needs to execute ``make`` at each directory to build both the eapp and
+the host.
+
+``OUTPUT_DIR`` is the path to the output files. Since we started from :doc:`Quick Start
```

Comment:
+ This is more Vault specific, but people will want to see the
   binaries locally. The 'output dir' should be an extra copy by
   default, rather than the only output.

Reply:
This is "vault.sh" section, so I assume it is OK..? OUTPUT_DIR only appears in this section and "packaging enclave" section. 

BTW, the expectation was to make people read through the script and figure out what each line is doing so that they can understand the whole build process.
If we were to just make people use the script we wrote, again, it'd be better to have a top-level script and high-level documentation for the script (e.g., just explaining the usage).

Reply:
I meant that the change suggested is not to the documentation, but to the vault script.

Reply:
sorry I misunderstood, you meant to change the name of the variable?